src/lib/src/data-access/push-notifications/hooks/use-push-notifications.ts

```diff
@@ -0,0 +1,70 @@
+import * as Notifications from 'expo-notifications';
+import { useNavigationContainerRef, useRootNavigationState } from 'expo-router';
+import { useEffect, useState, useRef } from 'react';
+import { AppState, AppStateStatus } from 'react-native';
+import { ObtainPushNotificationsTokenArgs, pushNotificationsService } from '../service';
+
+export interface UsePushNotificationsArgs extends ObtainPushNotificationsTokenArgs {
+  subscribeDevice: ({ expoToken }: { expoToken: string }) => void;
+  isAuthenticated: boolean;
+  onNotificationResponse?: (notification: Notifications.Notification) => void;
+}
+
+export const usePushNotifications = ({
```

Comment:
I'm not sure if we should add this hook to the package. Because the implementation can be very different in different projects (like Docky, for example).